code/controllers/subsystems/jobs.dm

```diff
@@ -370,11 +328,13 @@ SUBSYSTEM_DEF(job)
 	H.job = rank
 
 	if(!joined_late)
-		var/turf/S = pickSpawnLocation(job.type)
-		if(istype(S))
-			H.forceMove(S)
+		var/obj/S = get_roundstart_spawnpoint(rank)
+
+		if(istype(S, /obj/landmark/join/start) && istype(S.loc, /turf))
+			H.forceMove(S.loc)
 		else
-			LateSpawn(H.client, rank)
+			var/datum/spawnpoint/spawnpoint = get_spawnpoint_for(H.client, rank)
+			H.forceMove(pick(spawnpoint.turfs))
```

Comment:
Why spawn code changes was reverted?

Reply:
Unfortunately i can't know all changes that you did so i assume by default that most of the code is behind. I can revert it back but does it really worth it except freeing some memory ?